docs/kms_secretlock.md

```diff
@@ -0,0 +1,143 @@
+# Aries New Secure KMS, SecretLock and Crypto capabilities
+
+The aries-framework-go project has a mandate to protect keys from being compromised during usage or storage of keys. 
+This goal can achieved by hiding raw key bytes from being exchanged between the user and the framework. Also, keys
+should be secured and locked by a master key that is knwon to the user and protected by a passphrase. In order 
+to achieve this, the following criteria must be met:
+ - The KMS service should support providing a key reference (or keyset handle) whenever a key is required. It must not
+   return or accept raw contents of keys within the framework and with the user.
+ - A SecretLock service is needed to protect keys by encrypting them when being stored.
+ - The SecretLock service must work with a master key that can be provided by the user.
+ - The master key should also be encrypted to protect it from being compromised.
+ - It should be unlocked by a passphrase known to the user.
+ - Finally the crypto services must only accept key references provided by the KMS to execute operations.
+ 
+There are several ways to prepare and use the new KMS and SecreatLock services. The below sections describe each path:
+
+## No SecretLock/KMS options
+
+For a user who doesn't have a master key created, they can create an Aries instance without SecretLock/KMS options
+which will create a new unsecured master key behind the scenes and save it to local disk. This key will be used to 
```

Comment:
Doesn't work for WASM.

Reply:
I think this section of text should be deferred to another PR once we figure out defaults for each target.

Reply:
Updated with a more relevant comment.

Reply:
created #1358 to add a default js-worker secretLock
and #1359 to add a default agent-rest secretLock